src/Chapter3.hs

```diff
@@ -910,6 +1036,23 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = 
+  Gold {
+          unGold :: Int
+       } deriving (Show)
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append g1 g2 = Gold (unGold g1 + unGold g2)
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append = (++)
+
+instance (Append a) => Append (Maybe a) where
+  append :: Maybe a -> Maybe a -> Maybe a
+  append (Just x) (Just y) = Just (append x y)
+  append _ _               = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality, it would be more beneficial is this function returned `Just` when at least one argument is `Just`.

This is how the real instance of `Maybe` works for the similar typeclass 🙂